test/one/profiler/test/SourceLocator.java

```diff
@@ -0,0 +1,68 @@
+package one.profiler.test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class SourceLocator {
+    public static String tryGetFrom(Throwable e, int ignoreFrames) {
+        StackTraceElement[] stackTrace = e.getStackTrace();
+
+        if (stackTrace.length > ignoreFrames) {
+            StackTraceElement element = stackTrace[ignoreFrames];
+            String fileName = element.getFileName();
+            int lineNumber = element.getLineNumber();
+            String className = element.getClassName();
+
+            try {
+                Class<?> clazz = Class.forName(className);
+                String filePath = getFilePathFromClassLoader(fileName, clazz);
+
+                return getSourceCodeLines(filePath, lineNumber);
+            } catch (ClassNotFoundException ex) {
+                return "Error: Class not found - " + className;
+            }
+        }
+        return "No stack trace available";
+    }
+
+    private static String getFilePathFromClassLoader(String fileName, Class<?> clazz) {
+        String packagePath = clazz.getPackage().getName().replace('.', File.separatorChar);
```

Comment:
I'd just use `/` for better readability. It works even on Windows.

Reply:
It also makes sense to cut everything after the last `$` in case it's a nested class.